### PR TITLE
Relational API basics for R client

### PR DESCRIPTION
**Pull request by hannes**

### CODE REVIEW

**Review by Mytherin — APPROVED** [review]
LGTM, I think this can be merged

**Comment by hannes** [comment]
I'm taking out a lot of special handling for function creation for now, going to handle those with macros instead.

File: tools/rpkg/R/Result.R [hunk 1]
```diff
@@ -63,7 +63,7 @@ list_to_df <- function(x) {
     return(x)
   }
   attr(x, "row.names") <- c(NA_integer_, -length(x[[1]]))
-  class(x) <- "data.frame"
+  class(x) <- c( "data.frame")
```
Comment (krlmlr) [inline tools/rpkg/R/Result.R:66]: A glitch?

```suggestion
  class(x) <- "data.frame"
```

File: tools/rpkg/R/backend-dbplyr__duckdb_connection.R [hunk 2]
```diff
@@ -14,9 +14,6 @@
 #' x <- tibble(txt = c("why", "video", "cross", "extra", "deal", "authority"))
 #' dbx <- copy_to(con, x, overwrite = TRUE)
 #'
-#' # x %>% filter(grepl("^.[hrx]", txt))
-#' dbx %>% filter(grepl("^.[hrx]", txt))
```
Comment (krlmlr) [inline tools/rpkg/R/backend-dbplyr__duckdb_connection.R:18]: Do we have an idea what causes the re2 valgrind warnings?

**Review by krlmlr — APPROVED** [review]

File: tools/rpkg/R/backend-dbplyr__duckdb_connection.R (hunk 2 above)
Comment (hannes) [reply tools/rpkg/R/backend-dbplyr__duckdb_connection.R:18]: Yes but lets discuss in person

**Pull request merged**